From 0314ef5451fa949648c606ec1b560f8c91c39729 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Mon, 11 Jan 2016 10:58:23 +0100
Subject: [PATCH] dracut: set pipefail in the final initramfs creation step

so any error in the last creation step is caught and dracut errors out,
if e.g. there isn't enough space.

Resolves: rhbz#1252449
---
 dracut | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/dracut b/dracut
index cb79dbf..adad6f3 100755
--- a/dracut
+++ b/dracut
@@ -333,9 +333,10 @@ if strstr "$modules_loaded" " fips " && command -v prelink >/dev/null; then
 fi
 
 type pigz &>/dev/null && gzip=pigz || gzip=gzip
-if ! ( umask 077; cd "$initdir"; find . |cpio -R 0:0 -H newc -o --quiet| \
+if ! ( set -o pipefail; umask 077; cd "$initdir"; find . |cpio -R 0:0 -H newc -o --quiet| \
     $gzip -9 > "$outfile"; ); then
     derror "dracut: creation of $outfile failed"
+    rm -f "$outfile"
     exit 1
 fi
 dinfo "Wrote $outfile"
